File: chatdownloader/src/_types/leaderboardtypes.py
"""
Publically accessible leaderboard types
"""
from _types import BadgeInformation
from typing import Optional
from dataclasses import dataclass
from dataclasses_json import DataClassJsonMixin

DEFAULT_ELO = 1200

@dataclass
class LeaderboardExportItem(DataClassJsonMixin):
    id: str
    rank: int
    elo: float
    username: str
    delta: float
    avatar: str
    badges: Optional[list[BadgeInformation]]


@dataclass
class LeaderboardInnerState:
    id: str
    username: str
    avatar: str
    badges: Optional[list[BadgeInformation]] = None
    previous_rank: Optional[int] = None
    elo: float = DEFAULT_ELO
    score: int = 0

File: chatdownloader/src/_types/leaderboardtypes.rs
/*
Publically accessible leaderboard types
*/

use serde::{Deserialize, Serialize};
use super::clptypes::BadgeInformation;

#[derive(Deserialize, Serialize, Clone, Debug)]
pub struct LeaderboardExportItem {
    pub id: String,
    pub rank: u16,
    pub elo: f32,
    pub username: String,
    pub delta: u16,
    pub avatar: String,
    pub badges: Option<Vec<BadgeInformation>>
}

#[derive(Deserialize, Serialize, Debug, Clone)]
pub struct LeaderboardInnerState {
    pub id: String,
    pub username: String,
    pub avatar: String,
    pub badges: Option<Vec<BadgeInformation>>,
    pub previous_rank: Option<u16>,
    pub elo: f32,
    pub score: f32
}